internal/app/upstream/client.go

```diff
@@ -289,3 +291,19 @@ func shutDown(ctx context.Context, conn *grpc.ClientConn) {
 func (e *UnsupportedResourceError) Error() string {
 	return fmt.Sprintf("Unsupported resource typeUrl: %s", e.TypeURL)
 }
+
+func updateConnectivityMetric(ctx context.Context, conn *grpc.ClientConn, scope tally.Scope) {
+	connectedStat := "connected"
+	for {
+		isChanged := conn.WaitForStateChange(ctx, conn.GetState())
+		// Based on the grpc implementation, isChanged is false only when ctx expires
+		// https://github.com/grpc/grpc-go/blob/0f7e218c2cf49c7b0ca8247711b0daed2a07e79a/clientconn.go#L509-L510
+		// We set an unusually high value in case of ctx expires to show the connection is dead.
+		if !isChanged {
+			scope.Gauge(connectedStat).Update(100)
```

Comment:
Should this be 0 to follow envoy's 0 and 1 connected states?

Reply:
I thought about it. In the previous commits in this PR i made 0 as a representative of ctx expire and made mappings for each other enum. This complexity was due to the fact that the `connectivity.State` enum started at 0. I preferred to tradeoff the extra code with the assumption to map grpc connectivity state with the gauge value to keep the code readable and use an abnormally high number for ctx cancellation state.